lib/blockchain/transaction.rs

```diff
@@ -0,0 +1,69 @@
+use crypto::{digest::Digest, sha2::Sha256};
+use std::collections::HashSet;
+
+/// Since the output structure is the same as the input structure and outputs are
+/// eventually used as inputs we just need to declare an output struct.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct Output {
```

Comment:
Putting this into a struct is a good idea, however I'm not really in love with the name. Output in this context doesn't really fit into any pattern which makes it need a fat comment when referenced.

Possible rename could be `TransactionData`, or otherwise (naming is hard!). This change if done well will make the following transaction code 10x readable.